### PR TITLE
feat(CB2-14234): Update smc-proh to remove snowball sqs pattern

### PR DESCRIPTION
**Pull request by cb-cs**
## Update smc-proh to remove snowball sqs pattern

Refactor to allow partial batch failure and remove snowball sqs pattern.

Updated code slightly to remove redundancy and make code more readable.

[CB2-14234](https://dvsa.atlassian.net/browse/CB2-14234)

## Checklist
- [X] Code has been tested manually
- [X] PR title includes the JIRA ticket number
- [X] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/handler.ts [hunk 1]
```diff
@@ -17,48 +17,32 @@ const handler = async (
...
-        if (dynamoDBEvent){
-          const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
+        const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
 
-          if (mcRequests != null && mcRequests.length > 0) {
-            await sendMCProhibition(mcRequests);
-          } else {
-            logger.info(`No relevant MC test results found in the record: ${JSON.stringify(dynamoDBEvent)}`);
-          }
+        if (mcRequests.length > 0) {
```
Comment (naathanbrown) [inline src/handler.ts:31]: are we sure mcRequests can't return null or undefined, means this would error needlessly

File: src/handler.ts [hunk 2]
```diff
@@ -17,48 +17,32 @@ const handler = async (
...
-      'Function not triggered, Missing or not true environment variable present',
-    );
-    callback(
-      null,
-      'Function not triggered, Missing or not true environment variable present',
-    );
+    logger.info('Function not triggered, Missing or not true environment variable present');
   }
+
+  callback(null, { batchItemFailures });
```
Comment (naathanbrown) [inline src/handler.ts]: can simplify this just down to be a return statement I think. Using callback is not needed and I think a bit old school these days

Comment (cb-cs) [reply src/handler.ts]: Yes, I mean I am not a fan of these callbacks either, I just didn't want to change too much logic, as testing will be more than just around the way the SQS/failures are handled then. There would be a bit of scope creep and a bit more testing, but happy to look at removing those if everyone is in favour?

File: src/handler.ts (hunk 1 above)
Comment (cb-cs) [reply src/handler.ts:31]: extractMCTestResults should be returning an empty array if nothing is found. Looks like the only non-stardard return is the 400 HttpError that is thrown if there is an issue when validating the McRequest.

File: src/handler.ts (hunk 2 above)
Comment (naathanbrown) [reply src/handler.ts]: I think it should be fine, you've changed that this function returns anyway already. Regression testing on this needs to include a successful run anyway so this should be covered already

Comment (cb-cs) [reply src/handler.ts]: Yes fair point, will remove and push now.

**Review by naathanbrown — APPROVED** [review]

**Review by matthew2564 — APPROVED** [review]

**Review by Daniel-Searle — APPROVED** [review]

File: src/handler.ts [hunk 3]
```diff
@@ -9,56 +9,40 @@ import { MCRequest } from './utils/MCRequest';
 const handler = async (
   event: SQSEvent,
   _context: Context,
-  callback: Callback,
+  _callback: Callback,
 ) => {
   let { NODE_ENV, SERVICE, AWS_REGION, AWS_STAGE, SEND_TO_SMC } = process.env;
```
Comment (m-mullen) [inline src/handler.ts]: I know this is pedantic and not a change you've made, but can this be a const?

Comment (cb-cs) [reply src/handler.ts]: Yes, done and pushed.

**Review by m-mullen — APPROVED** [review]

**Pull request merged**